docs/sdk/authentication/payment-delegation-database.md

```diff
@@ -0,0 +1,143 @@
+### `getUsers`
+
+[Link to implementation](https://github.com/LIT-Protocol/lit-assets/blob/f73b7e2a3c982fa8fe3a13ed816e07fa6e04fbd8/blockchain/contracts/contracts/lit-node/PaymentDelegation/PaymentDelegationFacet.sol#L53-L55)
+
+This `view` functions returns a list of all the `user`s a `payer` has delegated to.
+
+```js
+function getUsers(address payer) public view returns (address[] memory)
+```
+
+### `undelegatePayments`
+
+[Link to implementation](https://github.com/LIT-Protocol/lit-assets/blob/f73b7e2a3c982fa8fe3a13ed816e07fa6e04fbd8/blockchain/contracts/contracts/lit-node/PaymentDelegation/PaymentDelegationFacet.sol#L73-L80)
+
+This function removes the `user` as a delegatee for the `payer` (`payer` will be `msg.sender`).
+
+```js
+function undelegatePayments(address user) public
+```
+
+### `setRestriction`
+
+[Link to implementation](https://github.com/LIT-Protocol/lit-assets/blob/f73b7e2a3c982fa8fe3a13ed816e07fa6e04fbd8/blockchain/contracts/contracts/lit-node/PaymentDelegation/PaymentDelegationFacet.sol#L100-L104)
+
+This function takes a [LibPaymentDelegationStorage.Restriction](https://github.com/LIT-Protocol/lit-assets/blob/753ee03852d537bbcdd420ac19c132118bcca726/blockchain/contracts/contracts/lit-node/PaymentDelegation/LibPaymentDelegationStorage.sol#L15-L18) and sets the restriction to all `user`s for the `payer` (`payer` is `msg.sender`).
+
+```js
+function setRestriction(
+    LibPaymentDelegationStorage.Restriction memory r
+) public
+```
+
+### `getRestriction`
+
+[Link to implementation](https://github.com/LIT-Protocol/lit-assets/blob/f73b7e2a3c982fa8fe3a13ed816e07fa6e04fbd8/blockchain/contracts/contracts/lit-node/PaymentDelegation/PaymentDelegationFacet.sol#L57-L61)
+
+The getter returns the [LibPaymentDelegationStorage.Restriction](https://github.com/LIT-Protocol/lit-assets/blob/753ee03852d537bbcdd420ac19c132118bcca726/blockchain/contracts/contracts/lit-node/PaymentDelegation/LibPaymentDelegationStorage.sol#L15-L18) set by the `payer` for all their delegatee `user`s.
+
+```js
+function getRestriction(
+    address payer
+) public view returns (LibPaymentDelegationStorage.Restriction memory)
+```
+
+### `delegatePaymentsBatch`
+
+[Link to implementation](https://github.com/LIT-Protocol/lit-assets/blob/f73b7e2a3c982fa8fe3a13ed816e07fa6e04fbd8/blockchain/contracts/contracts/lit-node/PaymentDelegation/PaymentDelegationFacet.sol#L82-L87)
+
+Does the same as [delegatePayments](#delegatepayments), but accepts a list of `user`s as input.
+
+```js
+function delegatePaymentsBatch(address[] memory users) public
+```
+
+### `undelegatePaymentsBatch`
+
+[Link to implementation](https://github.com/LIT-Protocol/lit-assets/blob/f73b7e2a3c982fa8fe3a13ed816e07fa6e04fbd8/blockchain/contracts/contracts/lit-node/PaymentDelegation/PaymentDelegationFacet.sol#L89-L98)
+
+Does the same as [undelegatepayments](#undelegatepayments), but accepts a list of `user`s as input.
+
+```js
+function undelegatePaymentsBatch(address[] memory users) public
+```
```

Comment:
In general this looks good as an `api doc` but we will need supporting examples of how users will get 'delegation signatures ` for example as from this api doc it is non obvious.

Reply:
sorry what do you mean by delegation signatures?  i don't think the payment delegation contract system needs those.